Mage.Sets/src/mage/cards/r/RottenmouthViper.java

```diff
@@ -0,0 +1,164 @@
+        this.subtype.add(SubType.SNAKE);
+        this.power = new MageInt(6);
+        this.toughness = new MageInt(6);
+
+        // As an additional cost to cast this spell, you may sacrifice any number of nonland permanents. This spell costs {1} less to cast for each permanent sacrificed this way.
+        Cost cost = new SacrificeXTargetCost(StaticFilters.FILTER_CONTROLLED_PERMANENT_NON_LAND, true);
+        cost.setText("you may sacrifice any number of nonland permanents. " +
+                "This spell costs {1} less to cast for each permanent sacrificed this way");
+        this.getSpellAbility().addCost(cost);
+        this.addAbility(new SimpleStaticAbility(Zone.ALL, new RottenmouthViperCostReductionEffect()));
+
+        // Whenever Rottenmouth Viper enters or attacks, put a blight counter on it. Then for each blight counter on it, each opponent loses 4 life unless that player sacrifices a nonland permanent or discards a card.
+        Ability ability = new EntersBattlefieldOrAttacksSourceTriggeredAbility(
+                new AddCountersSourceEffect(CounterType.BLIGHT.createInstance())
+        );
+        ability.addEffect(new RottenmouthViperEffect(new CountersSourceCount(CounterType.BLIGHT)));
+        this.addAbility(ability);
+    }
+
+    private RottenmouthViper(final RottenmouthViper card) {
+        super(card);
+    }
+
+    @Override
+    public RottenmouthViper copy() {
+        return new RottenmouthViper(this);
+    }
+}
+
+class RottenmouthViperEffect extends OneShotEffect{
+
+    DynamicValue blightCounterAmount;
+
+    RottenmouthViperEffect(DynamicValue blightCounterAmount) {
+        super(Outcome.LoseLife);
+        this.blightCounterAmount = blightCounterAmount;
+        this.staticText = "Then for each blight counter on it, each opponent loses 4 life unless that player sacrifices a nonland permanent or discards a card.";
+    }
+
+    private RottenmouthViperEffect(final RottenmouthViperEffect effect) {
+        super(effect);
+        this.blightCounterAmount = effect.blightCounterAmount.copy();
+    }
+
+    @Override
+    public RottenmouthViperEffect copy() {
+        return new RottenmouthViperEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null){
+            int repeat = blightCounterAmount.calculate(game, source, this);
+            for (int i = 1; i <= repeat; i++) {
+
+                for (UUID opponentId : game.getOpponents(source.getControllerId())) {
+                    Player opponent = game.getPlayer(opponentId);
+                    if (opponent != null) {
+                        int permanents = game.getBattlefield().countAll(StaticFilters.FILTER_PERMANENT_NON_LAND, opponentId, game);
+                        if (permanents > 0 && opponent.chooseUse(outcome, "Sacrifices a nonland permanent? (Iteration " + i + " of " + repeat + ")",
```

Comment:
AI works bad with default outcome (`Outcome.LoseLife`). It's good practice to use custom outcome for each inner choose.

Must use `Outcome.Benefit` so AI will use sacrifice instead loose life all the time. In ideal world it's better to insert more complex logic like if opponent life > 10 then use else not use, but Outcome.Benefit is minimum.

Reply:
Added AI hints by 84db421f6dde227d635de7d6a4b28f9a7df1861c